python/mxnet/symbol/symbol.py

```diff
@@ -2451,7 +2451,8 @@ def var(name, attr=None, shape=None, lr_mult=None, wd_mult=None, dtype=None,
     handle = SymbolHandle()
     check_call(_LIB.MXSymbolCreateVariable(c_str(name), ctypes.byref(handle)))
     ret = Symbol(handle)
-    attr = AttrScope._current.value.get(attr)
+    with AttrScope():
```

Comment:
we can include above `not hasattr` logic here too.

Reply:
That logic is already included in the context

Reply:
we can avoid a dict copy this way.

Reply:
fair point, I just thought it was cleaner that way and had better separation of concerns. I will update.